activerecord/lib/active_record/enum.rb

```diff
@@ -140,6 +140,7 @@ def deserialize(value)
       end
 
       def serialize(value)
+        return unless mapping.has_key?(value) || mapping.has_value?(value)
         mapping.fetch(value, value)
```

Comment:
Looks like we're attempting to implement `#cast` in serialize here? Do we want to `assert_valid_value(value)`?